src/v/cloud_storage/partition_manifest.h

```diff
@@ -574,6 +576,8 @@ class partition_manifest : public base_manifest {
     /// Collection of replaced but not yet removed segments
     replaced_segments_list _replaced;
     model::offset _last_offset;
+    // Note: always set this member with `set_start_offset` as it has cached
+    // values associated with it that need to be invalidated.
```

Comment:
It's better, IMO, to enforce this via the type/access control system rather than hoping every modifier reads this. E.g., 

So something like:

```
class cached_offset {
  model::offset offset;
  mutable std::optional<kafka::offset> _cached_start_kafka_offset_local;
};
```

then either a friend definition restricted to the method which updates it, or push the update logic into the helper struct.